### ISSUE TITLE
Odd UI issue when not vaulting an invalid card after transaction fails

### ISSUE THREAD

**Issue by rossiter10** [braintree/braintree-web-drop-in#598]
### General information

* SDK/Library version: 1.19.0, also tried latest 1.22.1
* Environment: Production
* Browser and OS: Chrome Version 80.0.3987.122 (Official Build) (64-bit)

### Issue description

Steps to reproduce (card verification is enabled on our production account):
1. Load dropin UI
2. Add new card ("Choose another way to pay")
3. Fill in bad card info
4. **Uncheck** the save card checkbox (we're using [allowVaultCardOverride](https://braintree.github.io/braintree-web-drop-in/docs/current/module-braintree-web-drop-in.html#~cardCreateOptions))
5. Submit order
6. Payment method seems to get tokenized successfully, but the transaction fails. At this point, the "new payment form fields" are replaced with a new "saved" method entry in the dropin UI (it doesn't really get saved to the account though).
7. At this point, the user can't edit that new card to fix the details, which could be confusing to the user. They'd have to add a new method again.

So I'm assuming the teardown/re-initialize would work, although that might be tricky with our integration.

I thought the [clearFieldsAfterTokenization ](https://braintree.github.io/braintree-web-drop-in/docs/current/module-braintree-web-drop-in.html#~cardCreateOptions)option might address this, but it didn't seem to work.

I think this might be related to this edge case [described here](https://github.com/braintree/braintree-web-drop-in/pull/481#issuecomment-466532561).

Thanks!

**Comment by crookedneighbor** [comment]
Yeah, I think we need to supply a method to remove unvaulted payment methods and navigate between for cases like this. 

Thanks for opening the issue. We're currently working on v2 of the Drop-in, so I'll mark this with that label so we know to address this behavior as we revamp how vaulting works in Drop-in.

**Comment by crookedneighbor** [comment]
Gonna just restate for reference what I think our plan should be in v2. For unvaulted payment methods in the context of a customer that has already vaulted payment methods, we should display a terminal view indicating that the payment is ready to be processed instead of a list of saved payment methods.

(as this is a breaking change to the behavior of the UI, we need to do this as part of the v2 work)

**Comment by rossiter10** [comment]
> in the context of a customer that has already vaulted payment methods

Just to clarify, I'm seeing this behavior even when the customer doesn't have any vaulted payment methods.

**Comment by crookedneighbor** [comment]
Yes, we need a way to remove the payment method regardless. It's just especially strange when it's stacked up with a bunch of other saved methods.

**Comment by rossiter10** [comment]
@crookedneighbor - Thank you for the responses. In the meantime, I was testing out calling clearSelectedPaymentMethod() in this scenario and that seems to kind of reset things and display the credit card fields again. Just curious what you thought about that and if there are different problems I should be aware of using that method. I think it did clear all the form values, which isn't too terrible. Thanks!

**Comment by crookedneighbor** [comment]
Yes, that should work in the flow you have. 

I don't think that will work if the customer has saved payment methods, since it'll just remove it from the list of payment methods and show the other ones instead of going back to the card view.

But it sounds like for your setup it should work just fine.
